extensions/interactions/PencilCodeEditor/PencilCodeEditor.html

```diff
@@ -38,3 +37,17 @@
 <script type="text/ng-template" id="shortResponse/PencilCodeEditor">
   <pre><[answerCode | truncateAtFirstLine]></pre>
 </script>
+
+<script type="text/ng-template" id="modals/confirmationModal">    
+  <div class="modal-header">
+    <button type="button" class="close" ng-click="close(false)" data-dismiss="modal" aria-hidden="true">&times;</button>
+    <h4 class="modal-title">Confirmation Required</h4>
+  </div>
+  <div class="modal-body">
+    <p>Are you sure you want to reset your code?</p>
+  </div>
+  <div class="modal-footer">
+    <button type="button" ng-click="close(false)" class="btn btn-default" data-dismiss="modal">No</button>
```

Comment:
These buttons could probably be named better: Reset code (and maybe make it red, since it is a dangerous operation); Cancel.

Can we also put all this text into the translation system?